File: 2026/07-July/21-Tuesday/solution.cs
public class Solution {
    public int MaxActiveSectionsAfterTrade(string s) {
        int ones = 0;
        int prevZero = 0;
        int best = 0;
        int n = s.Length;

        int i = 0;

        while (i < n) {
            int j = i;
            while (j < n && s[j] == s[i])
                j++;

            int len = j - i;

            if (s[i] == '1')
                ones += len;
            else {
                best = Math.Max(best, prevZero + len);
                prevZero = len;
            }

            i = j;
        }

        return ones + best;
    }
}

File: 2026/07-July/21-Tuesday/solution.py
class Solution:
    def maxActiveSectionsAfterTrade(self, s: str) -> int:
        ones = 0
        prev_zero = 0
        best = 0
        i = 0
        n = len(s)

        while i < n:
            j = i
            while j < n and s[j] == s[i]:
                j += 1

            length = j - i

            if s[i] == '1':
                ones += length
            else:
                best = max(best, prev_zero + length)
                prev_zero = length

            i = j

        return ones + best
